Tensile/KernelWriter.py

```diff
@@ -1090,6 +1169,139 @@ def setupNewTile(self, kernel, tensorParametersA, tensorParametersB, isPap, isOp
+              localReads.addText(self.comment("local read a"))
+              localReadCodeA, packCodeA = self.localReadDo(kernel, plrIdx*self.numIterPerCoalescedReadA, iui*self.numReadsIterCoalescedA, 0, tensorParametersA)
+              localReads.addCode(localReadCodeA)
+              pack[plrIdx*self.numIterPerCoalescedReadA].addCode(packCodeA)
+              localReads.addText(self.comment("local read b"))
+          if u < kernel["LoopIters"]/self.numIterPerCoalescedReadB - self.numItersPLR or (kernel["PrefetchGlobalRead"] and u > localWriteEndIter):
+            if iui*self.numReadsIterCoalescedB < kernel["InnerUnroll"]:
+              localReadCodeB, packCodeB = self.localReadDo(kernel, plrIdx*self.numIterPerCoalescedReadB, iui*self.numReadsIterCoalescedB, 0, tensorParametersB)
+              localReads.addCode(localReadCodeB)
+              pack[plrIdx*self.numIterPerCoalescedReadB].addCode(packCodeB)
+          if not isResetLroIter or iui != kernel["InnerUnroll"]-1:
+            if u < kernel["LoopIters"]/self.numIterPerCoalescedReadA - self.numItersPLR or (kernel["PrefetchGlobalRead"] and u > localWriteEndIter):
+              if iui*self.numReadsIterCoalescedA < kernel["InnerUnroll"]:
+                localReads.addText(self.comment("local read increment a"))
+                localReads.addText(self.localReadInc(kernel, iui, tensorParametersA))
+            if u < kernel["LoopIters"]/self.numIterPerCoalescedReadB - self.numItersPLR or (kernel["PrefetchGlobalRead"] and u > localWriteEndIter):
+              if iui*self.numReadsIterCoalescedB < kernel["InnerUnroll"]:
+                localReads.addText(self.comment("local read increment b"))
+                localReads.addText(self.localReadInc(kernel, iui, tensorParametersB))
+
+      if kernel["PrefetchGlobalRead"]:
+        if u == localWriteEndIter+1 or (u == (localWriteEndIter+1)%kernel["LoopIters"] and kernel["ScheduleIterAlg"] == 2):
+          if self.enable["Wait"]:
+            waitLWCode.addCode(self.wait(kernel, tensorParametersA, tensorParametersB, -1, 0, -1, \
+                "3wait for local write"))
+          if self.enable["Sync"]:
+            syncCode.addCode(self.syncThreads(kernel))
+
+      if isSwapAndResetLwoIter: # ResetLroIter
+        if kernel["PrefetchGlobalRead"]:
+          if self.enable["LocalWrite"]:
+            # local write for next iter, used to have local writes here
+            pointerLWCode.addText(self.comment("local write swap offsets a"))
+            pointerLWCode.addText(self.localWriteSwapOffsets(kernel, tensorParametersA))
+            pointerLWCode.addText(self.comment("local write swap offsets b"))
+            pointerLWCode.addText(self.localWriteSwapOffsets(kernel, tensorParametersB))
+            pointerLWCode.addText(self.localWriteInitPointers(kernel, tensorParametersA))
+            pointerLWCode.addText(self.localWriteInitPointers(kernel, tensorParametersB))
```

Comment:
if kernel["PrefetchGlobalRead"]: is checked just above, can avoid checking again here

Reply:
done